app/lib/rocketchat.js

```diff
@@ -996,6 +996,25 @@ const RocketChat = {
 			log(e);
 		}
 	},
+	async hasPermissionsByUserRoles(permissions = [], userRoles = []) {
```

Comment:
To avoid some [duplicated logic](https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/develop/app/lib/rocketchat.js#L978), try to extract some of this logic to another function that can be reused when we're looking for permissions on user roles or room+user roles.